src/service.py

```diff
@@ -280,16 +280,12 @@ def write_to_file(path: Path, content: str, mode: Optional[int] = None) -> bool:
     """Write to file with provided content."""
     success = True
     try:
-        logger.info("Writing file to %s.", path)
-        fileobj = (
-            os.fdopen(os.open(path, os.O_CREAT | os.O_WRONLY, mode), "w", encoding="utf-8")
-            if mode
-            # create file with default permissions based on default OS umask
-            else open(path, "w", encoding="utf-8")  # pylint: disable=consider-using-with
-        )
-        with fileobj as file:
+        with open(path, "w", encoding="utf-8") as file:
             file.write(content)
-    except (NotADirectoryError, PermissionError) as err:
+
+        if mode:
```

Comment:
```suggestion
        if mode is not None:
```

Technically we should compare with None here, because just checking for truthy value will fail if the user specifically sets `mode=0o000`. In practice though this doesn't make sense, so fine to leave as-is.